articles/stripe-integration-react/index.md

```diff
@@ -0,0 +1,139 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/stripe-integration-react/
+title: Integrating Stripe with React
+description: Learn how to create a Stripe account and integrate Stripe Payment with your React project.
+author: lalithnarayan-c
+date: 2020-09-02T00:00:00-10:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/stripe-integration-react/hero.jpg
+    alt: Stripe React Integration
+---
+All online stores need a payment gateway, irrespective of the service or product being sold. As such, it is important for developers to ensure the implementation is successful and that payments are secure for customers. In this article, we learn how to set up the Stripe API for all payment purposes. Stripe is the [go-to payment processor](https://stripe.com/en-in/payments/payment-methods-guide) for developers thanks to its developer-friendly API, highest grade of payment processing security, and detailed [documentation](https://stripe.com/docs/api).
+
+#### Integration With React
+Let's assume that we run an online store and have enough information to process the order: order details, order total, and user details. We will focus on the following:
+
+1. Creating a Stripe Account
+2. Switching to Test Mode on Stripe
+3. Obtaining Stripe Access Token from Dashboard
+4. Integrating with React using `react-stripe-checkout`
+
+Before we get started with the implementation, let's understand the flow of payment information.
+
+#### How does Stripe help us secure financial transactions?
+1. Card details are entered on the website.
+2. The web application sends a request to the Stripe API with card details in it.
+3. Stripe API stores the card details securely and sends a token to the web application.
+4. The token is a point of reference to identify the card. Note that the card details are never shared with anyone for security reasons. The backend server receives the token and other meta-data related to the purchase.
+5. The backend application makes a request to the API with the token in place. Also, it's a good practice to recalculate the price at the backend, as the token does not contain information about the total amount to be charged.  
```

Comment:
This line needs a little bit of clarification. Is this a second request to the Stripe API? Can you send the CC details and the price with the first API request to make this transaction more streamlined/faster? When was the first price calculation that you are now suggesting the reader recalculate?